### ISSUE TITLE
Fathom DevTools panel not recognizing inspected element?

### ISSUE THREAD

**Issue by Osmose** [mozilla/fathom-fox#13]
STR:

1. Visit https://www.amazon.com/Nintendo-Switch-Neon-Blue-Red-Joy/dp/B01MUAGZ49/ref=sr_1_2?ie=UTF8&qid=1535647997&sr=8-2&keywords=switch&th=1
2. Open the devtools and use the inspector to inspect a DOM element
3. Click the "Fathom" tab

The panel appears blank for me on the latest Nightly: 

![screen shot 2018-08-30 at 10 13 25 am](https://user-images.githubusercontent.com/193106/44867685-63ac0300-ac3d-11e8-8898-e8ecc1903732.png)

I see some CSP issues in my browser console:

```
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“style-src”).
fathomfox
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“style-src”).
fathomfox:1:1
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“style-src”). fathomfox 
```

I also get a weird error from "devtoolsPanel.js" about "Simmer being undefined", but the UUID in the error doesn't match FathomFox in about:support so it's probably not related?

**Comment by erikrose** [comment]
Chasing it now. Can't repro on 61.0.1 but can on the 2018-06-27 Nightly (63.0a1). I don't, however, see the CSP warnings.

We used to have Warnings/Errors/Network buttons on the console. I don't see them anymore. Did they go away for everyone, or am I uniquely hosed?

**Comment by erikrose** [comment]
Hmm. Oddly enough, it works great when loaded in debugging mode with `web-ext`. It fails only when installed from AMO. I tried both ways on a blank Nightly profile to be sure.

**Comment by erikrose** [comment]
Trying to substitute the `options = {useContentScriptContext: true}` arg of `browser.devtools.inspectedWindow.eval` for our injection but haven't had luck yet. CC @globau.

**Comment by erikrose** [comment]
Well that's cute. simmer.js is empty in the web-ext release artifact. Hmm.

**Comment by erikrose** [comment]
I released a quick fix—1.2.1—which solves the Simmer-is-missing problem. Now things work a bit better but still not perfectly: I get a lot of `false` in the Selector column, which makes the things that actually use it grumpy (resulting in console errors). STR:

1. Go to https://grinchcentral.com/.
2. Use the Inspector to select a paragraph.
3. Switch to the Fathom devpanel. Label it. It works fine.
4. If you were to select other paragraphs, those work too.
5. Select the "An Open Letter" headline.
6. Switch to the Fathom panel. It reads `false`. Hmm. We ought to be able to figure that out now.

**Comment by erikrose** [comment]
I think those CSP errors you saw were fairly innocuous ones complaining about our use of a few `style` attrs here and there in our UI. We should fix those, but it's unrelated to this issue. What I cannot find documented is whether WebExtensions' insertions into a page are exempt from the page's CSP.

**Comment by erikrose** [comment]
One test is worth a thousand docs. I tried the devpanel on https://github.com/, which has a pretty locked down CSP for scripts. It worked. So it appears CSP is not a problem for us. The remaining weirdness to track down is [the grinchcentral false values](https://github.com/mozilla/fathom-fox/issues/13#issuecomment-417510604).

**Comment by globau** [comment]
looks like a problem with simmer itself; `Simmer($0)` on that element returns false, which we're not gracefully handling.

**Comment by globau** [comment]
we need to increase the max depth, which defaults to 3.  that element is `ASIDE.body:nth-child(2) > ARTICLE > H1.entry-title > A`.

**Comment by erikrose** [comment]
This should all be fixed in 1.2.2, which is now up on AMO. Good teamwork, all! :-D
